pravega/pravega
# Issue-3239: make DYNAMIC_LOGGER non-static to avoid test code race condition

Signed-off-by: kevinhan88 <kevinhan88@gmail.com>

**Change log description**  
Made DYNAMIC_LOGGER of AppendProcessor and PravegaRequestProcessor non-static,  hence we don't have to overwrite static variable in test code which may cause race condition when concurrent tests run

**Purpose of the change**  
Fixes #3239

**What the code does**  
Changed DYNAMIC_LOGGER of AppendProcessor and PravegaRequestProcessor from static to non-static, and added setter for it. 
In the test code, DYNAMIC_LOGGER is set up with mock object, which is of instance scope, to prevent race condition.

**How to verify it**  
Unit tests passed. 
Need to observe further at Travis to ensure no race condition to happen.
